### DISCUSSION TITLE
Authenticate with a PAT on the `/autoimport` calls

### DISCUSSION THREAD

**Discussion by abrondijk** [firefly-iii/firefly-iii#7629] [Firefly III]
Is there a way to authenticate yourself on the `/autoimport` call with a PAT?

It seems as though the only way to automate autoimports, is by having the `FIREFLY_III_ACCESS_TOKEN` environment variable set in the data-importer container. This makes it rather inconvenient to automate autoimports for multiple users, as it would mean I'd have to have a data-importer container running for each and every user.

**Comment by JC5** [comment]
No, you will have to run multiple containers or start/stop the container before you import.

**Reply by abrondijk** [reply]
Is this a design choice?

**Reply by JC5** [reply]
Yes. The PAT is a sensitive piece of data and this makes it easier to keep the data importer as stateless as possible.

**Reply by abrondijk** [reply]
I get that it's a sensitive piece of data, but now it requires the admin to have both the config.json _and_ the PAT of the user to automate nordigen imports.

Besides, wouldn't be more stateless if the data-importer didn't have to know the PAT beforehand? I suppose that it would be the most stateless if the data-importer only had to know the nordigen ID/secret, and provide the config.json/PAT through a post request.

Or is there perhaps a better way to automate nordigen imports than by using the `/autoimport` endpoint?

**Reply by JC5** [reply]
You could use the docker container with environment variables and import directly, over the command line.
